objects/src/transaction/proven_tx.rs

```diff
@@ -383,16 +382,16 @@ impl Deserializable for TxAccountUpdate {
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct InputNoteCommitment {
     nullifier: Nullifier,
-    note_id: Option<NoteId>,
+    note_hash: Option<Digest>,
 }
```

Comment:
An alternative to this would be to define `InputNoteCommitment` as:
```rust
pub struct InputNoteCommitment {
    nullifier: Nullifier,
    note_hash: Option<NoteHeader>,
}
```
(and we can implement `NoteHeader::note_hash()` to compute note hash from note ID and metadata)

The main advantage of doing this is that this won't require a separate index by note hash in the node (i.e., we can still look up notes by ID to check if they are present in the chain). cc @polydez  in case I'm not being correct here.

The disadvantage is including extra 32 bytes per note with delayed authentication - but I'm thinking this is not too bad.

The above change could affect only the `InputNoteCommitment` struct. The MASM code and `ToInputNoteCommitments` could probably remain as they are now.

Reply:
> The main advantage of doing this is that this won't require a separate index by note hash in the node (i.e., we can still look up notes by ID to check if they are present in the chain)

Are you talking about database index? This seems to be a concern of the protobuf specification, and not the protocol. Is this struct used for that too?

Reply:
This struct is used in the `ProvenTransaction` struct which is what gets submitted to the node with the `SubmitProvenTransaction` request. The node then needs to figure out if any of the notes with delayed authentication are already in the database. Currently, it does this by looking up notes by their ID - but if the ID is not there, we need a separate mechanism to look up notes by their hash.

So, the options we have:
1. Use `NoteHeader` in `InputNoteCommitment` as I suggested above.
2. Modify `ProvenTransaction` in some other way to contain note Ids for notes with delayed authentication.
3. Modify `SubmitProvenTransaction` endpoint to take more info in addition to the `ProvenTransaction` struct.
4. Add an index to the notes table in `miden-node` to enable fast lookup by note hash.

I think out of the 4 options the first one is preferred.

Reply:
I've implemented this in 38120515c543542a7dc41241c3898618d9b5f826.